pkg/pillar/devicenetwork/dnc.go

```diff
@@ -705,6 +708,9 @@ func (ctx *DeviceNetworkContext) doUpdatePortConfigListAndPublish(
 	}
 	newplace, newIndex := lookupPortConfig(ctx, *current)
 	if newplace == nil {
+		// Current Got deleted.
+		// XXX - Why are we prioritizing index 0 here just because
+		// it was working before??
```

Comment:
If [0] was working we stick to it, otherwise we restart looking through the list. Hence deleting something which is > 0 and not currently used doesn't cause any disruption.

Reply:
Updated the comments with u r explanation 👍